src/storage/storage_manager.cpp

```diff
@@ -216,6 +237,16 @@ void SingleFileStorageManager::LoadDatabase(StorageOptions storage_options) {
 			}
 		}
 
+		if (storage_options.block_header_size.IsValid()) {
```

Comment:
On the storage front - only newer DuckDB versions can read data with an increased block header size (as mentioned by Carlo). When writing a database with a different block header size (as is the case when specifying encryption) - we should always write the new storage version to the file so that older DuckDB versions get a clear error when opening a file written with encryption enabled.